### PR TITLE
hotfix: wrong registry entry

### PR DESCRIPTION
**Pull request by noahjoeris**
- fix weth address
- add more error handling
- change balance fetching logic a bit

### CODE REVIEW

File: app/src/hooks/useTransfer.tsx [hunk 1]
```diff
@@ -52,146 +47,146 @@ const useTransfer = () => {
...
+            amount,
+            BigInt(0),
+          )
 
-          if (sendResult.failure) {
+          if (plan.failure) {
+            console.error('Validation failed:', plan)
             addNotification({
-              message: 'This transfer failed!',
+              message: plan.failure.errors[0].message,
```
Comment (Victor-Salomon) [inline app/src/hooks/useTransfer.tsx:71]: plan.failure.errors[0].message can't be undefined?

maybe plan.failure.errors.length? plan.failure.errors[0].message : 'This transfer failed!'

File: app/src/hooks/useErc20Balance.tsx [hunk 2]
```diff
@@ -28,31 +20,25 @@ const useErc20Balance = ({ network, token, address, context }: UseBalanceParams)
...
+          fetchedBalance = await fetchEthereumBalance(context, token, address)
+          console.log(token.name)
+          console.log('fetchedBalance', fetchedBalance)
           break
         }
 
         case Network.Polkadot: {
-          if (!context?.polkadot.api) return
           fetchedBalance = await fetchAssetHubBalance(context, token, address)
+          console.log(token.name)
```
Comment (Victor-Salomon) [inline app/src/hooks/useErc20Balance.tsx]: can we remove the logs? on the file?

Comment (noahjoeris) [reply app/src/hooks/useErc20Balance.tsx]: already done yeah

File: app/src/hooks/useTransfer.tsx (hunk 1 above)
Comment (noahjoeris) [reply app/src/hooks/useTransfer.tsx:71]: good point

**Review by Victor-Salomon — APPROVED** [review]
LGTM 🔥
More Sentry alerts 👌

**Pull request merged**